src/main.test.ts

```diff
@@ -0,0 +1,591 @@
+  runId: number;
+  headRef: string;
+  ref: string;
+  eventName: string;
+  actor: string;
+  serverUrl: string;
+}
+
+let mock: MockObj;
+
+const mockCore = core as jest.Mocked<typeof core>;
+const mockCache = cache as jest.Mocked<typeof cache>;
+jest.mock('got');
+const mockGot = mocked(got);
+const mockFn = jest.fn();
+mockGot.post = mockFn.bind({});
+
+const slackUrl = 'https://hooks.slack.com';
+const slackPath = '/services/test/test';
+
+nock(slackUrl).persist().post(slackPath).reply(200);
+
+/**
+ *
+ */
+function setupMock() {
+  jest.clearAllMocks();
+
+  mock = {
+    // Default action inputs
+    inputs: {
+      'current-status': 'success',
+      'slack-webhook': `${slackUrl}${slackPath}`,
+      'github-token': `${process.env.GITHUB_TOKEN}`,
+    },
+    repo: {
+      owner: 'reside-eng',
+      repo: 'workflow-status-slack-notification',
+    },
+    workflow: 'Failure workflow (for test purpose only)',
+    prNumber: 4,
+    runId: 23456,
+    headRef: 'main',
+    ref: 'main',
+    eventName: 'pull_request',
+    actor: 'workflowactor',
+    serverUrl: 'https://github.com',
+  };
+
+  mockCore.getInput.mockImplementation(
+    (name: string): string => mock.inputs[name] || '',
+  );
+
+  mockCache.restoreCache.mockImplementation(
+    async (): Promise<string | undefined> => {
+      try {
+        await fsp.readFile('last-run-status', 'utf8');
+        return 'test';
+      } catch (err) {
+        return undefined;
+      }
+    },
+  );
+
+  // Setting to this Github repo by default
+  context.payload.repository = {
+    name: mock.repo.repo,
+    owner: {
+      login: mock.actor,
+    },
+    clone_url: mock.repo,
+  };
+  context.workflow = mock.workflow;
+  context.runId = mock.runId;
+  context.payload.pull_request = {
+    number: mock.prNumber,
+    head: {
+      ref: mock.headRef,
+    },
+  };
+  context.ref = mock.ref;
+  context.eventName = mock.eventName;
+  context.actor = mock.actor;
+  context.serverUrl = mock.serverUrl;
+}
+
+/**
+ *
+ */
+async function writeStatusToCache() {
+  fsp.writeFile('last-run-status', `completed/failure`, {
+    encoding: 'utf8',
+  });
+}
+
+/**
+ *
+ */
+async function cleanCache() {
+  fsp.unlink('last-run-status');
+}
+
+// Test using cache only to retrieve previous status (re-run workflow behavior)
+describe('last run status retrieved from cache (re-run workflow behavior)', () => {
+  beforeEach(() => setupMock());
+
+  beforeAll(() => writeStatusToCache());
+
+  afterAll(() => cleanCache());
+
+  it('should send success notification if last run failed and current succeeded', async () => {
+    await run();
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: success');
+    expect(mockCore.info).toHaveBeenCalledWith('Success notification');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/failure',
+    );
+    expect(mockFn.mock.calls.length).toBe(1);
```

Comment:
```suggestion
    expect(mockFn.mock.calls.length).toStrictEqual(1);
```

This is a stronger assertion.